### PR TITLE
[scheduler] Enhancement: Update group functions

### PR DESCRIPTION
**Pull request by carvalhof**
This PR updates `next_runnable_group` and `get_offsets_for_ready_tasks` functions to avoid some memory allocations.

### CODE REVIEW

**Review by ppenna — CHANGES_REQUESTED** [review]
Blocking merge to be able to run benchmarks.

File: src/rust/runtime/scheduler/scheduler.rs [hunk 1]
```diff
@@ -228,15 +228,23 @@ impl Scheduler {
...
+            self.groups[self.current_group_id.into()].get_offsets_for_ready_tasks(&mut offsets);
+            if !offsets.is_empty() {
+                self.current_ready_tasks = offsets;
                 return;
             }
             // If we reach this point, then we have looped all the way around without finding any runnable tasks.
             if self.current_group_id == starting_group_index {
                 return;
             }
+
```
Comment (iyzhang) [inline src/rust/runtime/scheduler/scheduler.rs:245]: Thanks for catching this!

File: src/rust/runtime/scheduler/group.rs [hunk 2]
```diff
@@ -149,18 +149,16 @@ impl TaskGroup {
...
-            // Turn into bit iter.
-            let mut offset: Vec<InternalId> = BitIter::from(notified)
-                .map(|x| Self::get_pin_slab_index(i, x).into())
-                .collect();
-            result.append(&mut offset);
+            let notified: u64 = self.waker_page_refs[waker_page_index].take_notified();
+            // Grab notified waker pages.
+            for waker_page_offset in BitIter::from(notified) {
+                let offset: InternalId = Self::get_pin_slab_index(waker_page_index, waker_page_offset).into();
+                offsets.push(offset);
```
Comment (iyzhang) [inline src/rust/runtime/scheduler/group.rs:159]: I believe this changes the behavior into a lifo queue of the ready to run tasks. Is there a way that we can do this and still keep it fifo?

**Review by iyzhang — CHANGES_REQUESTED** [review]

**Comment by iyzhang** [comment]
Microbenchmarks (these microbenchmarks from PR #1198 run 1024 coroutines that are all runnable)
| | run_any | run_any_background |
|dev |1627 ns +/- 8 | 1451 ns +/- 5|
|this PR|1665 ns +/- 19 | 1574 ns +/-17|

**Pull request closed without merging**